src/retasc/product_pages_api.py

```diff
@@ -0,0 +1,140 @@
+            },
+        )
+        res.raise_for_status()
+        return res.json()
+
+    @lru_cache(maxsize=None)
+    @staticmethod
+    def release_schedules_by_flags(
+        self,
+        release_short_name: str,
+        flag_cond: str = "flags_or__in",
+        flags: str = "rcm,releng,exd,sp",
+        fields: str = "release_shortname,name,date_start,date_finish",
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}{release_short_name}/schedule-tasks/?{flag_cond}={flags}&fields=a,b,c
+        Only one flag in condition use `flags_in=sp`
+        Multiple flags in OR condition use `flags_or__in=sp,rcm,releng`
+        Multiple flags in AND condition use `flags_and__in=sp,ga`
+        The response have the following fields:release_shortname,name,slug,date_start,date_finish
+        """
+        pp_release_url = self.api_url + "releases/"
+        res = self.session.get(
+            f"{pp_release_url}{release_short_name}/schedule-tasks",
+            params={
+                flag_cond: flags,
+                "fields": fields,
+            },
+        )
+        res.raise_for_status()
+        return res.json()
+
+    @lru_cache(maxsize=None)
+    @staticmethod
+    def release_schedules_by_name_search(
+        self,
+        release_short_name: str,
+        condition: str,
+        fields: str = "release_shortname,name,slug,date_start,date_finish",
+    ) -> list[dict]:
+        """
+        https://{pp_release_url}{release_short_name}/schedule-tasks?name__regex={condition}&fields=a,b,c
+        The response have the following fields:release_shortname,name,slug,date_start,date_finish
+        """
+        pp_release_url = self.api_url + "releases/"
+        res = self.session.get(
+            f"{pp_release_url}{release_short_name}/schedule-tasks",
+            params={
+                "name__regex": condition,
+                "fields": fields,
+            },
+        )
+        res.raise_for_status()
+        return res.json()
+
+    @lru_cache(maxsize=None)
+    @staticmethod
+    def get_all_active_releases_tasks_for_product(self, product_shortname: str) -> dict:
+        """
+        Get all release names that are still supported by rest api:
+        https://{pp_release_url}?product__shortname={product_shortname}&phase__lt=1000&fields=shortname
+        Loop through the releases and get the tasks of it by rest api
+        https://{pp_release_url}{release_short_name}/schedule-tasks/fields=name,slug,date_start,date_finish
+        """
+
+        product_shortname = product_shortname.strip()
+        if product_shortname not in constants.SUPPORTED_PRODUCTS:
+            raise ValueError(
+                f"Product - {product_shortname} is not supported yet. The products we supported currently are {constants.SUPPORTED_PRODUCTS}"
+            )
+
+        releases_list = ProductPagesApi.releases_by_product_phase(
+            self, product_shortname
+        )
+        all_tasks = {}  # type: dict
```

Comment:
```suggestion
        all_tasks: dict = {}
```

A fuller type hint (still incomplete) is `dict[str, dict[str, list[dict]]]` which indicates that the dict would be somewhat difficult to use. It might be better to wrap it in a smarter object.

Reply:
ack